tests/compile-fail/ptr_int_cast.rs

```diff
@@ -2,7 +2,7 @@ fn main() {
     let x = &1;
     // Casting down to u8 and back up to a pointer loses too much precision; this must not work.
     let x = x as *const i32;
-    let x = x as u8; //~ ERROR a raw memory access tried to access part of a pointer value as raw bytes
+    let x = x as u8; //~ ERROR Miri evaluation error: ptr to u8 cast 
```

Comment:
This seems odd. What does the full message look like?

Reply:
`5:13: 5:20: Miri evaluation error: ptr to u8 cast [E0080]`
I'm recompiling because of the last change I did to rustc, after that I'll send you the full error when running `./miri run tests/run-pass/intptrcast.rs`

Reply:
That's an error message regression, if it still is the case with the latest patch I'll have to r- this while we figure that you.

Reply:
I believe that is happening because here
```rust
        match dest_layout.ty.sty {
            // Casting to a reference or fn pointer is not permitted by rustc,
            // no need to support it here.
            RawPtr(_) => Ok(ptr.into()),
            Int(_) | Uint(_) => {
                let size = self.memory.pointer_size();

                match self.force_bits(Scalar::Ptr(ptr), size) {
                    Ok(bits) => self.cast_from_int(bits, src_layout, dest_layout),
                    Err(_) if dest_layout.size == size => Ok(ptr.into()),
                    Err(e) => Err(e),
                }
            }
            _ => return err!(Unimplemented(format!("ptr to {:?} cast", dest_layout.ty))),
        }
```
Is matching with `_` at the end but I'm not sure why.

Reply:
Yeah that's what it sounds like.

Reply:
Apparently this was my fault when running the tests somehow. I double checked and the old error is dispatched when running `./miri test`